src/Microsoft.OData.Core/JsonLight/ODataJsonLightWriter.cs

```diff
@@ -786,13 +786,15 @@ protected override void StartNestedResourceInfoWithContent(ODataNestedResourceIn
             {
                 // Write @odata.context annotation for navigation property
                 var containedEntitySet = this.CurrentScope.NavigationSource as IEdmContainedEntitySet;
-                if (containedEntitySet != null)
+                if (containedEntitySet != null && this.jsonLightOutputContext.MessageWriterSettings.LibraryCompatibility == ODataLibraryCompatibility.Version6)

```

Comment:
>ODataLibraryCompatibility [](start = 124, length = 25)

what should we do for the case of setting.LibraryCompatiblity being Version7 or Latest?

Reply:
We shouldn't write context Url for nav props unless the contextUrl doesn't follow convention; and we currently have no way for the app to tell us that it doesn't follow convention, so we should just not write it.

---
In reply to: [175302067](https://github.com/OData/odata.net/pull/1109#discussion_r175302067) [](ancestors = 175302067)